ipatests/test_integration/test_authselect.py

```diff
@@ -0,0 +1,197 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+
+"""
+Module provides tests to verify that the authselect code works.
+"""
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_plugins.integration import tasks
+
+default_profile = 'sssd'
+preconfigured_profile = 'winbind'
+preconfigured_options = ['with-fingerprint']
+
+
+def check_authselect_profile(host, expected_profile, expected_options=[]):
+    """
+    Checks that the current authselect profile on the host
+    matches expected one
+    """
+    cmd = host.run_command(
+        ['cat', '/etc/authselect/authselect.conf'])
+    lines = cmd.stdout_text.splitlines()
+    assert lines[0] == expected_profile
+    options = lines[1::]
+    for option in expected_options:
+        assert option in options
+
+
+def apply_authselect_profile(host, profile, options=[]):
+    """
+    Apply the specified authselect profile and options with --force
+    """
+    cmd = ['authselect', 'select', profile]
+    cmd.extend(options)
+    cmd.append('--force')
+    host.run_command(cmd)
+
+
+class TestClientInstallation(IntegrationTest):
```

Comment:
It is possible to add an annotation to execute the test only on distributions supporting authselect
```
@pytest.mark.skipif(
    ipaplatform.NAME not in ['fedora', 'rhel', 'centos'],
    reason="Authselect is only available in fedora-like distributions")
```